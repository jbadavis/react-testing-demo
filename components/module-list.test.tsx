import {render} from '@testing-library/react';
import ModuleList from './module-list';

describe('ModuleList', () => {
  const mockModules = [
    {
      title: 'Foo',
      href: '/foo',
    },
    {
      title: 'Bar',
      href: '/bar',
    },
  ];

  let container: HTMLElement;

  beforeEach(() => {
    container = render(<ModuleList moduleList={mockModules} />).container;
  });

  test('renders correctly', () => {
    expect(container).toMatchSnapshot();
  });

  test('renders the correct number of items', () => {
    expect(container.querySelectorAll('li')).toHaveLength(mockModules.length);
  });
});
