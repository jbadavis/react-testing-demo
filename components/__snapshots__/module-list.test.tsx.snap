// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModuleList renders correctly 1`] = `
<div>
  <ol
    class="css-3p5u52"
    role="list"
  >
    <li
      class="css-0"
    >
      <a
        href="/foo"
      >
        <a
          class="chakra-link css-1157erc"
        >
          Foo
        </a>
        .
      </a>
    </li>
    <li
      class="css-0"
    >
      <a
        href="/bar"
      >
        <a
          class="chakra-link css-1157erc"
        >
          Bar
        </a>
        .
      </a>
    </li>
  </ol>
</div>
`;

exports[`ModuleList when a list of modules is provided renders correctly 1`] = `
<div>
  <ol
    class="css-3p5u52"
    role="list"
  >
    <li
      class="css-0"
    >
      <a
        href="/foo"
      >
        <a
          class="chakra-link css-1157erc"
        >
          Foo
        </a>
        .
      </a>
    </li>
    <li
      class="css-0"
    >
      <a
        href="/bar"
      >
        <a
          class="chakra-link css-1157erc"
        >
          Bar
        </a>
        .
      </a>
    </li>
  </ol>
</div>
`;

exports[`ModuleList when a list of modules isn't provided renders correctly 1`] = `
<div>
  <ol
    class="css-3p5u52"
    role="list"
  >
    <li
      class="css-0"
    >
      <a
        href="/introduction"
      >
        <a
          class="chakra-link css-1157erc"
        >
          Introduction
        </a>
        .
      </a>
    </li>
    <li
      class="css-0"
    >
      <a
        href="/snapshot-testing"
      >
        <a
          class="chakra-link css-1157erc"
        >
          Snapshot Testing
        </a>
        .
      </a>
    </li>
  </ol>
</div>
`;
